### DISCUSSION TITLE
What is the recommended way to convert DeltaQuantity to Quantity?

### DISCUSSION THREAD

**Discussion by benhutchison** [erikerlandson/coulomb#407] [General]
I have a bunch of millisecond timestamps modeled as `DeltaQuantity`s. 

I need to bin them into groups n seconds wide. That is, all timestamps within a time period of n seconds should be mapped to the same group (ie index).

To do this requires dividing the timestamp quantities by the window width. But I can't divide a DeltaQuantity by a Quantity directly. Rather, I ought to convert the DeltaQuantity into a Quantity. 

I found myself wanting a `toQuantity` operation for `DeltaQuantity` that yields a quantity using whatever origin/zero is defined naturally by that DeltaQuantity. Making a separate zero value and subtracting it from each timestamp feels like quite a lot of bother..

**Comment by benhutchison** [comment]
```scala
extension [VL, UL, B] (d: DeltaQuantity[VL, UL, B])
  def toQuantity: Quantity[VL, UL] = d.value.withUnit[UL]
```

**Reply by erikerlandson** [reply]
Your expression `d.value.withUnit[UL]`  is the way to do it now.  In the scala2 version, I actually supported a `.toQuantity` method, and I made the decision to *not* support it in scala3, because I felt like it violated the subtle but important semantic differences between DeltaQuantity and Quantity.  If you want to convert it, the "algebraically correct" way would probably be something like `d - zero`, where `zero` would be the DeltaQuantity for posix epoch for time, and zero Kelvin for Temperature.

I'm currently reluctant to provide it but I can think about it some.
